module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1267,12 +1264,15 @@ public function editlistAction()
             $listTags = $favoritesService
                 ->formatTagStringForEditing($tagsService->getListTags($list, $list->getUser()));
         }
+        $recordIds = $this->params()->fromQuery('ids', $this->params()->fromPost('ids', []));
+        $records = $this->getRecordLoader()->loadBatch($recordIds);
```

Comment:
Do you think it might be worth passing the $recordIds directly to the view model, and making the template smart enough to use $recordIds when set, and construct values using $records otherwise? That way we could avoid a potentially expensive extra Solr lookup.

I'm thinking the template logic could look something like:

```
  <?php
    if (!isset($this->recordIds)) {
      $this->recordIds = array_map(
        fn ($record) => $record->getSourceIdentifier() . '|' . $record->getUniqueId(),
        $this->records ?? []
      );
    }
  ?>
  <?php foreach ($this->recordsIds as $current): ?>
    <input type="hidden" name="ids[]" value="<?=$this->escapeHtmlAttr($current)?>">
  <?php endforeach; ?>
```

Reply:
Will adjust, also i think i had the testsuite installed but i will get it working :)

Reply:
Got the mink tests running this morning so will do changes

Reply:
noticed other problems with the implementation + tests, will try and adjust them also